src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticStartAnalysisScope.cs

```diff
@@ -383,22 +384,22 @@ public HostCompilationStartAnalysisScope(HostSessionStartAnalysisScope sessionSc
             _sessionScope = sessionScope;
         }
 
-        public override AnalyzerActions GetAnalyzerActions(DiagnosticAnalyzer analyzer)
+        public override AnalyzerActions? GetAnalyzerActions(DiagnosticAnalyzer analyzer)
         {
-            AnalyzerActions compilationActions = base.GetAnalyzerActions(analyzer);
-            AnalyzerActions sessionActions = _sessionScope.GetAnalyzerActions(analyzer);
+            AnalyzerActions? compilationActions = base.GetAnalyzerActions(analyzer);
+            AnalyzerActions? sessionActions = _sessionScope.GetAnalyzerActions(analyzer);
 
-            if (sessionActions.IsEmpty)
+            if (sessionActions.GetValueOrDefault(AnalyzerActions.Empty).IsEmpty)
```

Comment:
>GetValueOrDefault(AnalyzerActions.Empty) [](start = 31, length = 40)

IMO this would read cleaner as `?.`  and `== true`

Reply:
This is now simplified.